File: micro_services/microservices-client/src/services/MessageService.js
import Service from './Service';

const resource = '/message';

class MessageService extends Service{
    sendMessage(message) {
        return this.getApiClient().post(`${resource}`, message);
    }
    getConversation(requestId) {
        return this.getApiClient().get(`${resource}/${requestId}`);
    }
}


export default MessageService;

File: monolite/CarRental/src/main/java/com/programatori/carrental/service/MessageService.java
package com.programatori.carrental.service;

import com.programatori.carrental.dto.MessageDTO;
import org.springframework.http.ResponseEntity;

import java.text.ParseException;
import java.util.List;

public interface MessageService {

    public ResponseEntity<?> sendMessage(MessageDTO messageDTO) throws ParseException;
    public List<MessageDTO> getRequestConversation(Long requestId);
}
